scipy/integrate/odepack.py

```diff
@@ -225,6 +227,12 @@ def odeint(func, y0, t, args=(), Dfun=None, col_deriv=0, full_output=0,
         ml = -1  # changed to zero inside function call
     if mu is None:
         mu = -1  # changed to zero inside function call
+
+    dt = np.diff(t)
+    if not((dt >= 0).all() or (dt <= 0).all()):
+        raise ValueError("The values in t must be monotonically increasing "
```

Comment:
I guess, since we allow for repeated values, this should read as `must either all nondecreasing or all nonincreasing but not a mixture.` or something along those lines.

Reply:
By every definition that I have seen, the statement "t is monotonically increasing" means t[i] <= t[i+1] for all i, so repeated values are allowed.

Reply:
But I guess it wouldn't hurt to state explicitly that repeated values are allowed.

Reply:
I also thought that "monotonically increasing" was `<` and "monotonically nondecreasing was `<=` but upon reading the definitions this does indeed appear mistaken! I agree adding a `(t[i] <= t[i+1] for all i)` sort of parenthetical is worth it.